File: InterconnectBackend/Models/Requests/UpdateEntityPositionRequest.cs
using Models.Enums;

namespace Models.Requests
{
    public class UpdateEntityPositionRequest
    {
        public required int Id { get; set; }
        public required EntityType Type { get; set; }
        public required int X { get; set; }
        public required int Y { get; set; }
    }
}

File: InterconnectFrontend/src/api/requests/UpdateEntityPositionRequest.ts
import { EntityType } from "../../models/enums/EntityType.ts";

/**
 * Request object for updating the position of an entity in the simulation.
 */
interface UpdateEntityPositionRequest {
    /** ID of the entity to move */
    id: number;
    /** Type of the entity */
    type: EntityType;
    /** New X coordinate */
    x: number;
    /** New Y coordinate */
    y: number;
}

export default UpdateEntityPositionRequest;
